src/dynamic_sharing_service/policy/dynamic.policy.builder.js

```diff
@@ -0,0 +1,202 @@
+'use strict';
+
+const DynamicPolicy = require('./dynamic.policy');
+const WantedAttributeBuilder = require('./wanted.attribute.builder');
+const constants = require('../../yoti_common/constants');
+
+const SELFIE_AUTH_TYPE = 1;
+const PIN_AUTH_TYPE = 2;
+
+module.exports = class DynamicPolicyBuilder {
+  constructor() {
+    this.wantedAttributes = {};
+    this.wantedAuthTypes = [];
+  }
+
+  /**
+   * @param {WantedAttribute} wantedAttribute
+   */
+  addWantedAttribute(wantedAttribute) {
+    let key = wantedAttribute.getName();
+    if (wantedAttribute.getDerivation()) {
+      key = wantedAttribute.getDerivation();
+    }
+
+    this.wantedAttributes[key] = wantedAttribute;
+    return this;
+  }
+
+  /**
+   * @param {boolean} optional
+   * @param {string} name
+   */
+  withWantedAttribute(optional, name) {
```

Comment:
@echarrod as discussed, I'll change this method to `withWantedAttributeByName(name, optional)` (switch order of arguments) and have `optional` default to `false`.

`addWantedAttribute()` will be renamed to `withWantedAttribute()` to match Java SDK